### PR TITLE
Add a chapter on syntax rewriting using the Dotty compiler

### PR DESCRIPTION
**Pull request by eloots**
This adds a section on syntax rewriting of source code by the Dotty compiler

### CODE REVIEW

File: docs/dotty-syntax-rewrites.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+Converting source files one by one using the Dotty compiler, is time consuming
+(and requires you to specify the class path which can be a hassle). A more
+efficient way to do this in bulk is to recompile your code as part of your
+build. For example, if you have an sbt based build, simply add the required
+options to `Compile / scalacOptions` and do a clean compilation of your project.
+
+### Loss of formatting when cycling through syntax versions
+
+When formatting tools such as [scalafmt](https://scalameta.org/scalafmt) are used to apply customised formatting to your code, cycling back and forth between different Scala 3 syntax
+variants may result in differences when going full circle.
```
Comment (adpi2) [inline docs/dotty-syntax-rewrites.md:263]: In the Incompatibtility table page I have used the Blockquote syntax `>` to make a note box that is floating inside the text. It is like a tips and tricks box. ([result](https://scalacenter.github.io/scala-3-migration-guide/docs/incompatibility-table.html#scala-213-deprecations-and-feature-warnings)) ([source](https://github.com/scalacenter/scala-3-migration-guide/blob/master/docs/incompatibility-table.md#scala-213-deprecations-and-feature-warnings))

How about using the same layout for these two paragraphs. The first one could be floating inside the **Rewrite to SIB syntax** section. The second one could be floating inside the **Moving back to classic syntax**

What do you think?

**Review by adpi2 — CHANGES_REQUESTED** [review]

File: docs/dotty-syntax-rewrites.md (hunk 1 above)
Comment (eloots) [reply docs/dotty-syntax-rewrites.md]: Good idea: I'll update this accordingly.
BTW; when I read your comments in the review the first time (almost 10 days ago), I think you made a comment about adding an sample invocation of `dotc` as an example. I don't see this comment anymore. Was it removed or am I dreaming?

**Review by adpi2 — APPROVED** [review]
Thank you for this good piece of writing!

File: docs/dotty-syntax-rewrites.md (hunk 1 above)
Comment (adpi2) [reply docs/dotty-syntax-rewrites.md]: I have no recollection of such a comment :smile: 
I said on the slack channel that I like that you show the `dotc -help` output and so I did the same on the [Dotty migration mode page](https://scalacenter.github.io/scala-3-migration-guide/docs/dotty-rewrites.html)

**Pull request merged**